x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/20_search_application_put.yml

```diff
@@ -112,15 +124,57 @@ teardown:
           indices: [ "test-index1" ]
 
   - match: { error.type: "version_conflict_engine_exception" }
-
 ---
 "Create Search Application - Insufficient privilege":
   - do:
       catch: forbidden
       search_application.put:
-        name: another-search-application
+        name: another-test-search-application
+        create: true
+        body:
+          indices: [ "another-index" ]
+
+  - match: { error.type: "security_exception" }
+
+  # Check neither aliases nor Search Application were created
+  - do:
+      indices.exists_alias:
+        name: another-test-search-application
+
+  - is_false: ''
+
+  - do:
+      catch: missing
+      search_application.get:
+        name: another-test-search-application
+
+---
+"Update Search Application - Insufficient privilege":
+  - do:
+      search_application.put:
+        name: test-search-application
         create: true
+        body:
+          indices: [ "test-index1", "test-index2" ]
+
+  - do:
+      catch: forbidden
+      search_application.put:
+        name: test-search-application
         body:
           indices: [ "another-index" ]
 
   - match: { error.type: "security_exception" }
+
+  # Check aliases and previous Search Application still exist
+  - do:
```

Comment:
This test fails as of now - when the Search App is changed, but the alias fails to be updated (permissions in this case), then the Search App contains the changed config and not the original one.